sound/soc/sof/sof-priv.h

```diff
@@ -67,100 +67,110 @@ struct snd_sof_pdata;
  * and DSP device(s).
  */
 struct snd_sof_dsp_ops {
+
 	/* probe and remove */
-	int (*remove)(struct snd_sof_dev *sof_dev);
-	int (*probe)(struct snd_sof_dev *sof_dev);
+	int (*probe)(struct snd_sof_dev *sof_dev); /* mandatory */
+	int (*remove)(struct snd_sof_dev *sof_dev); /* optional */
 
 	/* DSP core boot / reset */
-	int (*run)(struct snd_sof_dev *sof_dev);
-	int (*stall)(struct snd_sof_dev *sof_dev);
-	int (*reset)(struct snd_sof_dev *sof_dev);
+	int (*run)(struct snd_sof_dev *sof_dev); /* mandatory */
+	int (*stall)(struct snd_sof_dev *sof_dev); /* optional */
+	int (*reset)(struct snd_sof_dev *sof_dev); /* mandatory */
 	int (*core_power_up)(struct snd_sof_dev *sof_dev,
-			     unsigned int core_mask);
+			     unsigned int core_mask); /* optional */
 	int (*core_power_down)(struct snd_sof_dev *sof_dev,
-			       unsigned int core_mask);
-
-	/* pre/post firmware run */
-	int (*pre_fw_run)(struct snd_sof_dev *sof_dev);
-	int (*post_fw_run)(struct snd_sof_dev *sof_dev);
-
-	/* DSP PM */
-	int (*suspend)(struct snd_sof_dev *sof_dev, int state);
-	int (*resume)(struct snd_sof_dev *sof_dev);
-	int (*runtime_suspend)(struct snd_sof_dev *sof_dev, int state);
-	int (*runtime_resume)(struct snd_sof_dev *sof_dev);
-
-	/* DSP clocking */
-	int (*set_clk)(struct snd_sof_dev *sof_dev, u32 freq);
+			       unsigned int core_mask); /* optional */
 
 	/* Register IO */
 	void (*write)(struct snd_sof_dev *sof_dev, void __iomem *addr,
-		      u32 value);
-	u32 (*read)(struct snd_sof_dev *sof_dev, void __iomem *addr);
+		      u32 value); /* mandatory */
+	u32 (*read)(struct snd_sof_dev *sof_dev,
+		    void __iomem *addr); /* mandatory */
 	void (*write64)(struct snd_sof_dev *sof_dev, void __iomem *addr,
-			u64 value);
-	u64 (*read64)(struct snd_sof_dev *sof_dev, void __iomem *addr);
+			u64 value); /* mandatory */
+	u64 (*read64)(struct snd_sof_dev *sof_dev,
+		      void __iomem *addr); /* mandatory */
```

Comment:
this is inconsistent with the fact that those functions return 0 if not supported?